core/templates/command_queue_mt.h

```diff
@@ -114,6 +114,7 @@ class CommandQueueMT {
 	uint32_t sync_awaiters = 0;
 	WorkerThreadPool::TaskID pump_task_id = WorkerThreadPool::INVALID_TASK_ID;
 	uint64_t flush_read_ptr = 0;
+	std::atomic<bool> pending;
```

Comment:
We'd rather use `SafeFlag` here. We only use `std::atomic` directly where we need tighter control.

Aside, but even more imporant, note that `std::atomic::store()` and `load()` use total sequential order, which issues much stricter memory barriers, being a potential big pessimization. `SafeFlag` is more relaxed, as it uses acquire-release.